docs/implementation.adoc

```diff
@@ -0,0 +1,507 @@
+== Implementation Guide
+
+This document provides the guidelines for a correct optimal implementation of the SCM API.
+
+=== Introduction
+
+The SCM API consists of a number of extension points:
+
+* The original `hudson.scm.SCM` extension point is responsible for:
+
+** performing checkouts from source control into a `FilePath`
+
+** calculating the changelogs between revisions
+
+** constructing the changelog parser
```

Comment:
I mean that changelog could be generate for diff in different ways - following parents that you so like to see in GH, so it not enough to ask about start-end, some kind or type of changelog should be also passed.